Framework/CurveFitting/src/Functions/FunctionQDepends.cpp

```diff
@@ -0,0 +1,152 @@
+// Mantid Coding standars <http://www.mantidproject.org/Coding_Standards>
+
+// Main Module Header
+#include "MantidCurveFitting/Functions/FunctionQDepends.h"
+// Mantid Headers from the same project
+// N/A
+// Mantid headers from other projects
+#include "MantidAPI/MatrixWorkspace.h"
+#include "MantidAPI/NumericAxis.h"
+#include "MantidKernel/UnitConversion.h"
+// third party libraries
+// N/A
+// standard library
+// N/A
+
+using Attr = Mantid::API::IFunction::Attribute;
+
+namespace {
+Mantid::Kernel::Logger g_log("FunctionQDepends");
+}
+
+namespace Mantid {
+namespace CurveFitting {
+namespace Functions {
+
+/* ===========
+   Public
+   ===========*/
+
+FunctionQDepends::FunctionQDepends() {
+  this->declareAttribute("Q", Attr(EMPTY_DBL()));
+  this->declareAttribute("WorkspaceIndex", Attr(EMPTY_INT()));
```

Comment:
To be honest, I didn't know of this method! When I started coding my first function I took as model one function that declared attributes in the constructor (there are few of these functions) and I went on from there. I will move the declarations to this method.